### PR TITLE
Add Reserved Words DB test pipeline for SQLite

### PR DESCRIPTION
**Pull request by juankx-bodo**
This database have a set of fields with Pydough and SQL reserved words. It has a set of relationships and tables to recreate multiple issue scenarios (Optimization errors, missing alias, alias name that is a reserved word)

The test with the issue about an alias using a reserved word is `keywords_cast_alias_and_missing_alias`. SQL files refsols weren't created because we are getting an `ERROR WHILE OPTIMIZING QUERY` message that wasn't happening with v1.00.09 or before. Affected tests with `ERROR WHILE OPTIMIZING QUERY` are `keywords_cast_alias_and_missing_alias` and `keywords_column_alias_reserved`
Metadata comments are missing, no meaninful data is here. The metadata file generated by the tool should be easily loaded here to run the tests and apply metadata changes if it fixes query errors. New test examples are expected to be added. 

Issues included on tests:
- [Invalid use of table alias name when table_name is a SQL reserved word](https://github.com/bodo-ai/PyDough/issues/432)
- [Invalid double escape to names already escaped](https://github.com/bodo-ai/PyDough/issues/433)
- [Invalid SQL when a column or table name includes a quoting character](https://github.com/bodo-ai/PyDough/issues/434)
- [SQL query optimization error regression](https://github.com/bodo-ai/PyDough/issues/435)

### CODE REVIEW

File: tests/test_pipeline_custom_datasets.py [hunk 1]
```diff
@@ -115,6 +115,145 @@
             ),
             id="wdi_albania_footnotes_1978",
         ),
+        #  uncomment and align to reproduce SQL query optimization failed. SQL generation works for SQLite
```
Comment (knassre-bodo) [inline tests/test_pipeline_custom_datasets.py:118]: This is not the right way to write tests:
- Do not comment out the code
- Instead, use skip markers inside the `pytest.param` (e.g., beneath `id=...`, put `marks=pytest.mark.skip`)
- You should not have 1 test covering multiple errors. Instead, have 1 `pytest.param` for each unique edge case you want to test. There should **NOT** be anything that needs to be uncommented in order to test things.

File: tests/test_pipeline_custom_datasets.py [hunk 2]
```diff
@@ -115,6 +115,145 @@
...
+                lambda: pd.DataFrame(
+                    {
+                        "description": ["One-One master row"],
+                    }
+                ),
+                "keywords_single_quote_use",
+            ),
+            id="keywords_single_quote_use",
+        ),
+        #  uncomment and align to reproduce SQL query optimization failed. SQL execution works for SQLite
```
Comment (knassre-bodo) [inline tests/test_pipeline_custom_datasets.py:164]: Same here

File: tests/test_sql_refsols/keywords_cast_alias_and_missing_alias_sqlite.sql [hunk 3]
```diff
@@ -0,0 +1,10 @@
+SELECT
+  cast.id2 AS id1,
+  cast.id AS id2,
```
Comment (knassre-bodo) [inline tests/test_sql_refsols/keywords_cast_alias_and_missing_alias_sqlite.sql:3]: These files should not exist if the tests have been skipped.

File: tests/test_pipeline_custom_datasets.py (hunk 1 above)
Comment (juankx-bodo) [reply tests/test_pipeline_custom_datasets.py:118]: Done!

File: tests/test_pipeline_custom_datasets.py (hunk 2 above)
Comment (juankx-bodo) [reply tests/test_pipeline_custom_datasets.py:164]: Done!

File: tests/test_metadata/reserved_words_graph.json [hunk 4]
```diff
@@ -0,0 +1,2202 @@
...
+        "original parent": "uppercase_master",
+        "original property": "lowercase_detail",
+        "singular": true,
+        "always matches": true,
+        "description": "reverse for relationship between uppercase_master and lowercase_detail",
+        "synonyms": []
+      }
+    ]
+  }
+]
```
[the remark this answers is not in the document]
Comment (juankx-bodo) [reply tests/test_metadata/reserved_words_graph.json:2202]: Done!

File: tests/test_pipeline_custom_datasets.py [hunk 5]
```diff
@@ -115,6 +115,166 @@
...
+                ),
+                "keywords_alias_reserved_word",
+            ),
+            id="keywords_alias_reserved_word",
+            marks=pytest.mark.skip(
+                """TODO: Issue: Invalid use of table alias name when table_name is a SQL reserved word
+                https://github.com/bodo-ai/PyDough/issues/432
+                The SQL query optimization error is a regression, it was not happening on v1.00.09 and before
+                https://github.com/bodo-ai/PyDough/issues/435
+            """
```
[the remark this answers is not in the document]
Comment (juankx-bodo) [reply tests/test_pipeline_custom_datasets.py:275]: Done!

File: tests/test_pipeline_custom_datasets.py [hunk 6]
```diff
@@ -115,6 +115,166 @@
...
+                        "__init__": [7051],
+                    }
+                ),
+                "keywords_python_sql_reserved",
+            ),
+            marks=pytest.mark.skip(
+                """TODO: Invalid SQL when a column or table name includes a quoting character
+                https://github.com/bodo-ai/PyDough/issues/434
+                The SQL query optimization error is a regression, it was not happening on v1.00.09 and before
+                https://github.com/bodo-ai/PyDough/issues/435
```
[the remark this answers is not in the document]
Comment (juankx-bodo) [reply tests/test_pipeline_custom_datasets.py:238]: Done!

File: tests/test_pipeline_custom_datasets.py [hunk 7]
```diff
@@ -115,6 +115,166 @@
...
+                """TODO: Issue: Invalid double escape to names already escaped in metadata
+                https://github.com/bodo-ai/PyDough/issues/433
+                The SQL query optimization error is a regression, it was not happening on v1.00.09 and before
+                https://github.com/bodo-ai/PyDough/issues/435
+            """
+            ),
+        ),
+        pytest.param(
+            PyDoughPandasTest(
+                r"""
```
[the remark this answers is not in the document]
Comment (juankx-bodo) [reply tests/test_pipeline_custom_datasets.py:205]: `r`removed from tests that don't need it.

File: tests/test_metadata/sf_masked_examples.json [hunk 8]
```diff
@@ -116,7 +116,7 @@
             "synonyms": ["customer id", "customer key", "customer number"]
           },
           {
-            "name": "_type",
+            "name": "loan_type",
```
Comment (juankx-bodo) [inline tests/test_metadata/sf_masked_examples.json:119]: The property with `"name": "type"` failed metadata validation. It was initially changed to `_type` to pass validation. Updating it to `loan_type`, which is a more meaningful property name.

File: pydough/errors/error_utils.py [hunk 9]
```diff
@@ -167,28 +168,122 @@ class ValidSQLName(PyDoughPredicate):
...
+    # Single-part unquoted SQL identifier (no dots here).
+    _UNQUOTED_SQL_IDENTIFIER = re.compile(r"^[A-Za-z_][A-Za-z0-9_]*$")
 
     def __init__(self):
         self.error_messages: dict[str, str] = {
             "identifier": "must have a SQL name that is a valid SQL identifier",
             "sql_keyword": "must have a SQL name that is not a reserved word",
         }
 
+    def _split_identifier(self, name: str) -> list[str]:
```
Comment (juankx-bodo) [inline pydough/errors/error_utils.py:180]: This change was necessary to pass validation for complex qualified SQL identifiers that contain quotes and reserved or escaped characters.

File: pydough/errors/error_utils.py [hunk 10]
```diff
@@ -167,28 +168,122 @@ class ValidSQLName(PyDoughPredicate):
     as the name for a SQL table path/column name.
     """
 
-    # Regex for unquoted SQL identifiers
-    _UNQUOTED_SQL_IDENTIFIER = re.compile(
-        r"^[A-Za-z_][A-Za-z0-9_]*(\.[A-Za-z_][A-Za-z0-9_]*)*$"
-    )
+    # Single-part unquoted SQL identifier (no dots here).
+    _UNQUOTED_SQL_IDENTIFIER = re.compile(r"^[A-Za-z_][A-Za-z0-9_]*$")
```
Comment (knassre-bodo) [inline pydough/errors/error_utils.py:172]: Don't prefix this one with `_`, and use a multiline string docstring beneath this kind of class variable, so that it will show up in tooltips

```suggestion
    UNQUOTED_SQL_IDENTIFIER = re.compile(r"^[A-Za-z_][A-Za-z0-9_]*$")
    """
    Regex pattern for a single-part unquoted SQL identifier (without dots).
    """
```

**Review by knassre-bodo — APPROVED** [review]

File: pydough/errors/error_utils.py (hunk 10 above)
Comment (juankx-bodo) [reply pydough/errors/error_utils.py]: Done!
